### DISCUSSION TITLE
Limiting Street View Source to Outdoor views only

### DISCUSSION THREAD

**Discussion by aancer-rca** [JustFly1984/react-google-maps-api#3279] [Q&A]
Hi,

I'm trying to limit my Google maps search results to only provide street view results that are outdoor views. 

I noticed that the following options are available for use by the JavaScript api: https://developers.google.com/maps/documentation/javascript/reference/street-view-service#StreetViewSource

To disable indoor street view they provide the following option: 
`{source: google.maps.StreetViewSource.OUTDOOR}` in the [StreetViewLocationRequest](https://developers.google.com/maps/documentation/javascript/reference#StreetViewLocationRequest)

However, I'm unsure how to implement this within the scope of the `react-google-maps-api` package? 

I would suspect that I could provide this option to the `StreetViewPanoramaOptions` but I do not see the option for a `source`. I attempted to manipulate the map `onLoad` and request a panorama with which I set on the map but it was not fruitful. 

Any ideas how I could set the OUTDOOR option with the react google map on the StreetViewPanorama?

**Comment by voidrlm** [comment]
You cannot limit the 'react google maps' as it does not provide a built-in option to set the street view panorama to an outdoor view directly by passing any parameters.
Define a React component that uses the "withGoogleMap" to render the Google Map with Street View and in this component, check the "location_type" field of the panorama data to determine if it's an outdoor view and do the needful if its an outdoor view or not
